Nosach/web-quiz-project/src/main/java/controllers/QuizController.java

```diff
@@ -0,0 +1,116 @@
+package controllers;
+
+import entity.Answer;
+import entity.Quiz;
+import entity.Subject;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.method.P;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.context.request.WebRequest;
+import service.QuizService;
+import service.SubjectService;
+import service.builder.QuizBuilder;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.web.bind.annotation.RequestMethod.DELETE;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+
+@Controller
+public class QuizController {
+
+    private static final Logger logger = Logger.getLogger(QuizController.class);
+
+    @Autowired
+    QuizService quizService;
+    @Autowired
+    SubjectService subjectService;
+    @Autowired
+    QuizBuilder quizBuilder;
+
+    @RequestMapping(method = GET, value = "list")
+    public String showQuizListForm(Model model){
+        List<Quiz> list =  quizService.getAllQuizzies();
+        model.addAttribute("quizzies", list);
+        return "list-of-quizzes";
+    }
+
+    @RequestMapping(method = GET, value = "add")
+    public String showQuizAddingForm(Model model){
+        List<Subject> subjects = subjectService.getAllSubjects();
+        model.addAttribute("subjects", subjects);
+        return "quiz-page";
+    }
+
+    @RequestMapping(method = POST, value = "add")
+    public String addQuizBasicInfo(HttpSession session, WebRequest req){
+        if(session.getAttribute("builder") == null){
+            quizBuilder.clean();
+            session.setAttribute("builder", quizBuilder);
+        }
+
+        String subject;
+        if (req.getParameter("subjectList") != ""){

```

Comment:
Maybe it's better, to pass request parameter as parameter in method?

Reply:
In this method I used three different request parametrs, so I desided not to separate them.